XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -397,13 +483,39 @@ private void ShowVcpuError(bool showAlways, bool testValue)
             }
         }
 
+        private long _prevVCPUsMax;
+
+        private void RefreshCurrentVCPUs()
+        {
+            // refresh comboBoxInitialVCPUs if it's visible and populated
+            if (comboBoxInitialVCPUs.Visible && comboBoxInitialVCPUs.Items.Count > 0)
+            {
+                if (_prevVCPUsMax == 0)
+                    _prevVCPUsMax = _OrigVCPUsMax;

```

Comment:
I think it would be clearer to initialise this when _OrigVCPUsMax is initialised.